lib/ansible/modules/network/nxos/nxos_bfd_global.py

```diff
@@ -0,0 +1,308 @@
+    ipv4_slow_timer:
+    ipv6_slow_timer:
+    description:
+      - does a thing
+    required: false
+  fabricpath-specific commands (only available for 5k/6k/7k)
+    fabricpath_interval:
+    description:
+      - does a thing
+    required: false
+    fabricpath_slow_timer:
+    description:
+      - does a thing
+    required: false
+    fabricpath_vlan:
+    description:
+      - does a thing
+    required: false
+'''
+EXAMPLES = '''
+- nxos_bfd_global:
+    echo_interface: Ethernet1/2
+    echo_interval: [50, 50, 3]
+    state: present
+'''
+
+RETURN = '''
+commands:
+    description: commands sent to the device
+    returned: always
+    type: list
+    sample: ["router bgp 65535", "vrf test", "router-id 192.0.2.1"]   # ***** TBD ****
+'''
+
+
+import re, yaml
+
+from ansible.module_utils.network.nxos.nxos import nxos_argument_spec, check_args
+from ansible.module_utils.network.nxos.nxos import load_config, run_commands, get_capabilities
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.network.common.config import CustomNetworkConfig
+from ansible.module_utils.connection import ConnectionError
+
+BFD_CMD_REF = '''
+# BFD does not support json format yet; use cli config strings instead.
+---
+echo_interface:
+  type: str
+  getval: (no )*bfd echo-interface *(\S+)*$
+  setval: '{0}bfd echo-interface {1}'
+  default: ''
+
+echo_rx_interval:
+  type: int
+  getval: bfd echo-rx-interval (\d+)$
+  setval: bfd echo-rx-interval {0}
+  default: 50
+  N3K: 250
+
+interval:
+  type: int_list
+  getval: bfd interval (\d+) min_rx (\d+) multiplier (\d+)
+  setval: bfd interval {0} min_rx {1} multiplier {2}
+  default: [50,50,3]
+  N3K: [250,250,3]
+
+slow_timer:
+  type: int
+  getval: bfd slow-timer (\d+)$
+  setval: bfd slow-timer {0}
+  default: 2000
+'''
+BFD_SHOW_CMD = "show run bfd all | incl '^(no )*bfd'"
+
+
+def init_cmd_ref(module):
+    '''Ensure BFD feature is enabled. Create cmd_ref dict.
+    '''
+    output = execute_show_command(module, 'show run bfd', 'text')
+    if output and 'CLI command error' in output:
+        if module.check_mode:
+            msg = "** 'feature bfd' is disabled. BFD will auto-enable when not in check_mode **"
+            module.warn(msg)
+        else:
+            load_config(module, 'feature bfd')
+
+    cmd_ref = yaml.load(BFD_CMD_REF)
+    cmd_ref = get_platform_defaults(module, cmd_ref)
+    return cmd_ref
+
+def get_platform_defaults(module, cmd_ref):
+    '''Get platform type and update cmd_ref with platform specific defaults
+    '''
+    cap = get_capabilities(module)
+    if cap:
+        device_info = cap['device_info']['network_os_platform']
+        plat = None
+        if 'N3K' in device_info:
+            plat = 'N3K'
+        if plat:
+            for k in cmd_ref:
+                if plat in cmd_ref[k]:
+                    cmd_ref[k]['default'] = cmd_ref[k][plat]
+
+    return cmd_ref
+
+
+def execute_show_command(module, command, format):
```

Comment:
Mind adding docstring here stating, among other things, what this returns (newline-separated str? list of str?)

Reply:
done